core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -444,10 +444,23 @@ final class ShuffleBlockFetcherIterator(
               throwFetchFailedException(blockId, address, e)
           }
 
-          input = streamWrapper(blockId, in)
+          var wrapCorruption: Boolean = false
```

Comment:
Wouldn't it be better to try to merge this block into the existing code below?

Something along the lines of:

```
try {
  val input = streamWrapper(blockId, in)
  if (detectCorrupt && !wrapCorruption && !input.eq(in) && size < maxBytesInFlight / 3) {
    // do the corruption detection
} catch {
  // existing catch cases from around L474
}
```

It seems you're trying to achieve the same things as the existing code (e.g. retry on IOException) so it would be nice to re-use the same code for that.

Reply:
Thanks for the review. I didn't do that to give more specific warning message and being able to point whether the problem happened during wrapping or after, but if there isn'r any value in separating that I can change it.

Reply:
The reason for the error is the same (corruption in the source data), so a different error message doesn't really add any value. The exception will contain the line where it happened so that's enough for debugging.

Reply:
yeah that is right. I will change that